src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlSearchType.cs

```diff
@@ -0,0 +1,28 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Search
+{
+    /// <summary>
+    /// Enum to specify the type of search to carry out.
+    /// </summary>
+    /// <remarks>
+    /// With the exception of None, the flags attribute requires enumeration constants to be powers of two, that is, 1, 2, 4, 8.
+    /// </remarks>
+    [Flags]
+    internal enum SqlSearchType
+    {
+        // Default value, set if we do not need to consider history or reindexing
+        None = 0,
+
+        // Set if we are including previous resource versions or deleted resources
+        History = 1,
```

Comment:
>1 [](start = 18, length = 1)

I think it's quite common practice to specify them via power of 2 like: 1<<0. 1<<1, 1<<2 and so on. Probably not necessary for 3 values enum.

Reply:
Oh nice, I like this idea - I think it would prevent someone from specifying `NewSearchType = 3` in the future (I put a comment in 9cdd203 but I think this is even more clear). Added in 1bbad9e!